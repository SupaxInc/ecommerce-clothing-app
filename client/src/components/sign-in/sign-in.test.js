import React from "react";
import { mount, shallow } from "enzyme";

import * as redux from 'react-redux';
import SignIn from "./sign-in";
import FormInput from "../form-input/form-input";

// Allows us to create mock stores multiple times if needed
export const createMockStore = ({ state, reducers }) => {
    const store = createStore(combineReducers(reducers), state);
    return {
        ...store,
        persistor: {
            persist: () => null
        }
    }
}

describe('Sign In component', () => {
    let wrapper;

    const useDispatchMock = jest.spyOn(redux, 'useDispatch');
    const dispatch = jest.fn();
    beforeEach(() => {
        useDispatchMock.mockImplementation(() => dispatch);
        wrapper = shallow(<SignIn />);
    });

    it('should render sign-in component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should submit form on click and dispatch', () => {
        expect.assertions(1);
        wrapper.find('form').simulate(
            'submit', {
                preventDefault() {}
            }
        );

        expect(dispatch).toHaveBeenCalled();
    });

    it('should update email state on change of email input', () => {
        // Mock useState hook for setCredentials function
        // const setCredentials = jest.fn();
        // const useStateMock = jest.spyOn(React, 'useState');
        // useStateMock.mockImplementation(() => setCredentials);

        // Check if form inputs exist
        expect(wrapper.find(FormInput).length).toBe(2);
        wrapper.find(FormInput).at(0).dive().find('.form-input').simulate('change', {
            target: {
                name: 'email',
                value: 'foo@gmail.com'
            }
        });

        expect(wrapper.find(FormInput).at(0).dive().find('.form-input').prop('value')).toEqual('foo@gmail.com');
        //expect(setCredentials).toHaveBeenCalled();
    });
}); 