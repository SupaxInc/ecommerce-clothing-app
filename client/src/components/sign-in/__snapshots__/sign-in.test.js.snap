// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In component should render sign-in component 1`] = `ShallowWrapper {}`;
